src/modules/limitOrders/services/safeBundleFlow/index.ts

```diff
@@ -91,6 +93,27 @@ export async function safeBundleFlow(
       { to: presignTx.to!, data: presignTx.data!, value: '0', operation: 0 },
     ]
 
+    const shouldZeroApprove = await shouldZeroApproveFn({
+      tokenContract: erc20Contract,
+      spender,
+      amountToApprove: inputAmount,
+      isBundle: true,
+    })
+
+    if (shouldZeroApprove) {
+      const zeroApproveTx = await buildZeroApproveTx({
+        erc20Contract,
+        spender,
+        currency: inputAmount.currency,
+      })
+      safeTransactionData.unshift({
+        to: zeroApproveTx.to!,
+        data: zeroApproveTx.data!,
+        value: '0',
+        operation: 0,
+      })
+    }
```

Comment:
all this logic was added for Limit Orders, did we add it for SWAP too? 
Also, if we do, will we duplicate logic?

Reply:
It is duplicated, as part of a separate PR. As @shoom3301 also mentioned probably we could make this abstracted, but at this stage not super sure how that would look like. (i.e. what is absolutely shared between limit and swap orders).